src/index.js

```diff
@@ -0,0 +1,217 @@
+  const popupCommentsEl = document.querySelector('.popup-comments');
+  const response = await fetch(`https://us-central1-involvement-api.cloudfunctions.net/capstoneApi/apps/71VEvRYFUclLOrVdlfg1/comments?item_id=item${data.id}`);
+  const commentsData = await response.json();
+  const commentTitleEl = document.querySelector('.comments-title');
+  commentTitleEl.textContent = `Comments (${commentsData.length})`;
+  commentsData.forEach((comment) => {
+    const commentEl = document.createElement('p');
+    commentEl.classList.add('popup-comment');
+    commentEl.textContent = `${comment.creation_date} ${comment.username}: ${comment.comment}`;
+    popupCommentsEl.appendChild(commentEl);
+  });
+
+  const commentForm = document.getElementById('popup-comment-form');
+  commentForm.addEventListener('submit', async (e) => {
+    e.preventDefault();
+    const name = commentForm.name.value;
+    const comment = commentForm.comment.value;
+    const commentData = {
+      item_id: `item${data.id}`,
+      username: name,
+      comment,
+    };
+
+    postComment(data.id, commentData);
+    commentForm.reset();
+    popupCommentsEl.innerHTML = '';
+    const response = await fetch(`https://us-central1-involvement-api.cloudfunctions.net/capstoneApi/apps/71VEvRYFUclLOrVdlfg1/comments?item_id=item${data.id}`);
+    const commentsData = await response.json();
+    const commentTitleEl = document.querySelector('.comments-title');
+    commentTitleEl.textContent = `Comments (${commentsData.length})`;
+    commentsData.forEach((comment) => {
+      const commentEl = document.createElement('p');
+      commentEl.classList.add('popup-comment');
+      commentEl.textContent = `${comment.creation_date} ${comment.username}: ${comment.comment}`;
+      popupCommentsEl.appendChild(commentEl);
+    });
+    const commentEl = document.createElement('p');
+    commentEl.classList.add('popup-comment');
+    commentEl.textContent = `${commentsData[1].creation_date} ${commentData.username}: ${commentData.comment}`;
+    popupCommentsEl.appendChild(commentEl);
+  });
+};
+
+const closePopup = (id) => {
+  const closeBtns = document.querySelectorAll('.popup-close-btn');
+  const commentsPopupWrapperEl = document.querySelector('.comments-popup-wrap');
+  const commentsPopupEl = document.querySelector(`#popup${id}`);
+  closeBtns.forEach((closeBtn) => {
+    closeBtn.addEventListener('click', () => {
+      commentsPopupEl.remove();
+      commentsPopupWrapperEl.style.display = 'none';
+    });
+  });
+};
+
+const renderCommentsPopup = (data) => {
+  const commentsPopupWrapperEl = document.querySelector('.comments-popup-wrap');
+  const commentsPopupEl = document.createElement('div');
+  commentsPopupEl.classList.add('comments-popup');
+  commentsPopupEl.id = `popup${data.id}`;
+  commentsPopupEl.innerHTML = `
+    <div class="popup-header">
+      <img class='popup-image' src='${data.image.original}'></img>
+      <span class='popup-close-btn'>&times;</span>
+    </div>
```

Comment:
- [ ] What if you could reduce the size of the popup image showing? currently, it's too big and hence making the modal look big and causing a long scroll to see other details. Kindly reduce the image size to improve the look and feel of your app 👍🏽

Reply:
fixed